pennylane/estimator/ops/op_math/controlled_ops.py

```diff
@@ -0,0 +1,2095 @@
+
+    The resources for this operation are computed using:
+
+    >>> qml.estimator.CH.resource_decomp()
+    [(2 x Hadamard), (2 x RY), (1 x CNOT)]
+
+    """
+
+    num_wires = 2
+
+    @property
+    def resource_params(self) -> dict:
+        r"""Returns a dictionary containing the minimal information needed to compute the resources.
+
+        Returns:
+            dict: Empty dictionary. The resources of this operation don't depend on any additional parameters.
+        """
+        return {}
+
+    @classmethod
+    def resource_rep(cls) -> CompressedResourceOp:
+        r"""Returns a compressed representation containing only the parameters of
+        the operator that are needed to compute the resources.
+
+        Returns:
+            :class:`~.pennylane.estimator.resource_operator.CompressedResourceOp`: A compressed representation of the operator.
+        """
+        return CompressedResourceOp(cls, cls.num_wires, {})
+
+    @classmethod
+    def resource_decomp(cls) -> list[GateCount]:
+        r"""Returns a list of ``GateCount`` objects representing the resources of the operator..
+
+        Resources:
+            The resources are derived from the following identities (as presented in this
+            `blog post <https://quantumcomputing.stackexchange.com/questions/15734/how-to-construct-a-controlled-hadamard-gate-using-single-qubit-gates-and-control>`_):
+
+            .. math::
+
+                \begin{align}
+                    \hat{H} &= \hat{R}_{y}(\frac{\pi}{4}) \cdot \hat{Z}  \cdot \hat{R}_{y}(\frac{-\pi}{4}), \\
+                    \hat{Z} &= \hat{H} \cdot \hat{X}  \cdot \hat{H}.
+                \end{align}
+
+            Specifically, the resources are defined as two ``RY``, two ``Hadamard`` and one ``CNOT`` gates.
+
+        Returns:
+            list[:class:`~.estimator.resource_operator.GateCount`]: A list of ``GateCount`` objects,
+            where each object represents a specific quantum gate and the number of times it appears
+            in the decomposition.
+        """
+        ry = resource_rep(RY)
+        h = resource_rep(Hadamard)
+        cnot = resource_rep(CNOT)
+        return [GateCount(h, 2), GateCount(ry, 2), GateCount(cnot, 1)]
+
+    @classmethod
+    def adjoint_resource_decomp(cls) -> list[GateCount]:
+        r"""Returns a list representing the resources for the adjoint of the operator.
+
+        Resources:
+            This operation is self-adjoint, so the resources of the adjoint operation results
+            are same as the originial operation.
+
+        Returns:
+            list[:class:`~.estimator.resource_operator.GateCount`]: A list of ``GateCount`` objects,
+            where each object represents a specific quantum gate and the number of times it appears
+            in the decomposition.
+        """
+        return [GateCount(cls.resource_rep())]
+
+    @classmethod
+    def controlled_resource_decomp(
+        cls,
+        num_ctrl_wires,
+        num_zero_ctrl,
```

Comment:
Mind just updating the type hinting across this file? 😅 
```suggestion
        num_ctrl_wires: int,
        num_zero_ctrl: int,
```